src/frontend/js/widgets/Dashboard/utils/teacherRoutes.tsx

```diff
@@ -32,35 +32,26 @@ export function getTeacherDashboardRoutes() {
           element: <TeacherDashboardCoursesLoader />,
         },
         {
-          path: getRoutePath(TeacherDashboardPaths.COURSE, {
-            courseId: ':courseId',
-          }),
+          path: getRoutePath(TeacherDashboardPaths.COURSE),
           children: [
             {
               index: true,
               element: (
                 <NavigateWithParams
-                  to={getRoutePath(TeacherDashboardPaths.COURSE_GENERAL_INFORMATION, {
-                    courseId: ':courseId',
-                  })}
+                  to={getRoutePath(TeacherDashboardPaths.COURSE_GENERAL_INFORMATION)}
```

Comment:
Not quit sure getRoutePath is useful anymore. Indeed, we could declare routes into an enum right now.

Reply:
It is useful only if we need to replace variables in the URL

Reply:
we can use [react-router generatePath](https://reactrouter.com/en/main/utils/generate-path) to replace route's params in path.